src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -217,22 +236,81 @@ private void handleStart(HttpExchange exchange) throws IOException {
         }
     }
 
+    private void handleStopOrUpdate(HttpExchange exchange, long id) throws IOException {
+        try {
+            JFRConnection conn =
+                    jfrConnectionToolkit.connect(
+                            jfrConnectionToolkit.createServiceURL("localhost", 0));
+            IFlightRecorderService svc = conn.getService();
+            IRecordingDescriptor dsc = svc.getAvailableRecordings().stream().filter(r -> r.getId() == id).findFirst().get();
+            RecordingOptionsBuilder builder = new RecordingOptionsBuilder(conn.getService());
+            
+            InputStream body = exchange.getRequestBody();
+            JsonNode jsonMap = mapper.readTree(body);
+            Iterator<Entry<String, JsonNode>> fields = jsonMap.fields();
+
+            while (fields.hasNext()) {
+                Map.Entry<String, JsonNode> field = fields.next();
+
+                switch(field.getKey()){
+                    case "state":
+                        if ("STOPPED".equals(field.getValue().toString())) {
+                            handleStop(exchange, id);
+                            break;
+                        } else if (!"UPDATE".equals(field.getValue().toString())) {
+                            exchange.sendResponseHeaders(HttpStatus.SC_BAD_REQUEST, BODY_LENGTH_NONE);
+                        }
+                        break;
+                    case "name":
+                        builder = builder.name(field.getValue().toString());
+                        break;
+                    case "duration":
+                        builder = builder.duration(field.getValue().asLong());
```

Comment:
If one of these fails, ex. `getValue().asLong()` fails because the client provided a non-`Long`-parseable value, I assume this throws an Exception of some kind? Do we map that out to a `BAD REQUEST` somewhere?

Reply:
no- I'll add that in

Reply:
according to [this](https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/JsonNode.html#asLong(long)) no exception is thrown and a default value is returned, which is not the behaviour we want, so I'll figure out a different way to convert the type